packages/babel-plugin-proposal-function-bind/src/index.ts

```diff
@@ -14,11 +14,17 @@ export default declare(api => {
     return scope.path.setData("functionBind", id);
   }
 
+  function getObject(bind: t.BindExpression) {
+    if (t.isExpression(bind.object)) {
+      return bind.object;
+    }
+
+    t.assertMemberExpression(bind.callee);
+    return bind.callee.object;
```

Comment:
If Babel parser already ensured `bind.callee` is a member expression, we can cast the type here and remove the runtime assertion. Hopefully when we refine the AST later, the cast will be redundant, too.
```suggestion
    return (bind.callee as t.MemberExpression).object;
```

We use runtime assertion for user-provided inputs, i.e. AST builders. If the AST can be trusted, in this case it was constructed from Babel parser, we can strip the runtime assertion.